docs/CODING_CONVENTIONS.md

```diff
@@ -0,0 +1,246 @@
+# Coding Conventions
+
+This document outlines coding conventions used in the Amethyst.
+
+We follow the [Rust API Guidelines].
+This document only cover topics which aren't already outlined there.
+
+[Rust API Guidelines]: https://rust-lang-nursery.github.io/api-guidelines/about.html
+
+## Error Handling
+
+#### Defining crate-local errors
+
+Custom errors _must_ be defined in a module called `error`.
+
+The error _must_ implement `std::error::Error`, which in turn requires `std::fmt::Display` and `std::fmt::Debug`).
+
+The `std::fmt::Display` implementation _must not_ format the wrapped error since this is already provided through
+`source` (see below).
+
+The error _should_ implement `std::fmt::Debug` through the `Debug` derive (see below), unless this is not supported.
```

Comment:
we should probably define what `should` and `must` mean but probably in a different document.

Reply:
Added a small section and linked to https://www.ietf.org/rfc/rfc2119.txt which is the inspiration for the keywords.